src/consensus/mod.rs

```diff
@@ -114,6 +155,16 @@ pub trait Consensus: Debug + Send + Sync + 'static {
     fn validate_header_parallel(&self, _: &BlockHeader) -> Result<(), DuoError> {
         Ok(())
     }
+
+    /// To be overridden for consensus validators' snap.
+    fn snapshot(
```

Comment:
For the `SnapRW`, it only read hardcoded `parliaSnapshot` for all dyn instance of SnapRW, so for every struct implement this func will only fetch parlia related snapshot.  Thus I don't think it's proper to define this func in general consensus. Or you can put the `SnapRW` outside parlia files and make it general.